File: frontend/src/types/classroom.ts
// Data models for classroom management

export interface Classroom {
  id: number
  name: string
  description?: string
  urlCount: number
  createdAt: string
}

export interface ClassroomUrl {
  id: number
  url: string
  urlType: string
  createdAt: string
}

export interface ClassroomDetail {
  id: number
  name: string
  description?: string
  createdAt: string
  urls: ClassroomUrl[]
}

// Request models
export interface CreateClassroomRequest {
  name: string
  description?: string
}

export interface AddUrlRequest {
  url: string
}

File: backend/src/main/kotlin/org/acme/Classroom.kt
package org.acme

import jakarta.persistence.*
import java.time.LocalDateTime

/**
 * Classroom entity represents a classroom in the system.
 * 
 * A classroom is a logical grouping representing a physical class or lesson group.
 * Each classroom can have multiple URLs associated with it for DNS filtering control.
 * 
 * This entity supports the Classroom URL Management feature, enabling teachers to
 * organize URL lists by class or subject.
 */
@Entity
@Table(name = "classrooms")
class Classroom {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    
    @Column(nullable = false, length = 100)
    var name: String = ""
    
    @Column(length = 500)
    var description: String? = null
    
    @Column(name = "created_at", nullable = false)
    var createdAt: LocalDateTime = LocalDateTime.now()
    
    @OneToMany(mappedBy = "classroom", cascade = [CascadeType.ALL], orphanRemoval = true)
    var urls: MutableList<ClassroomUrl> = mutableListOf()
}
